src/js/Setup.js

```diff
@@ -421,6 +500,32 @@ function setupAnimControls() {
             else button.classList.remove("CindyJS-active");
         }
     }
+
+    var speedDragging = false;
+
+    function speedDown(event) {
+        speedDragging = true;
+        speedDrag(event);
+    }
+
+    function speedDrag(event) {
+        if (!speedDragging) return;
+        var rect = slider.getBoundingClientRect();
+        var x = event.clientX - rect.left - slider.clientLeft + 0.5;
+        setSpeed(speedScale * x / rect.width + speedLo);
+    }
+
+    function speedUp(event) {
+        speedDragging = false;
+    }
+
+}
+
+var setSpeedKnob = null;
+
+function setSpeed(speed) {
+    simspeed = Math.max(0, speed); // do we want to allow negative speed?
```

Comment:
That's an interesting comment! I would love to be able to run simulations backwards.